gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java

```diff
@@ -507,6 +506,7 @@
   /**
    * Other configuration properties.
    */
+  public static final String DELIVERY_SEMANTICS = "delivery.semantics";
```

Comment:
Please make sure the property is namespaced, say, with "gobblin.runtime." Thus, we can avoid clashes with user-defined properties.